### PR TITLE
all env in one place

### PR DESCRIPTION
**Pull request by Mauricio-Natacci**
env var organized in one place only

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -32,6 +32,7 @@
     "@typescript-eslint/no-unsafe-return": "off",
     "@typescript-eslint/prefer-optional-chain": "off",
     "@typescript-eslint/no-misused-promises": "off",
-    "@typescript-eslint/unbound-method": "off"
+    "@typescript-eslint/unbound-method": "off",
+    "@typescript-eslint/prefer-nullish-coalescing": "off"
```
Comment (nathan-torquato) [inline .eslintrc]: I think you should not do this, nullish coalescing is such a great tool.

Why did you add this rule?

File: backend/config/index.ts [hunk 2]
```diff
@@ -2,13 +2,30 @@ import dotenv from 'dotenv'
...
+  environment: process.env.NODE_ENV,
+  jwtSecret: process.env.JWT_SECRET
 }
 
-// TODO: throw an error if the environment variable is not set
 if (!config.mongoURI) {
   throw new Error('MONGO_URI environment variable is not set')
 }
+
+if (!config.shouldServeReactApp) {
```
Comment (nathan-torquato) [inline backend/config/index.ts]: Either you didn't test the application locally or you set this to be "true" in your `.env` file, both options are wrong lol

File: backend/db/connectDb.ts [hunk 3]
```diff
@@ -1,18 +1,15 @@
...
-import dotenv from 'dotenv'
-
-dotenv.config()
 
 let mongoURI: string
-if (process.env.MONGO_URI) {
-  mongoURI = process.env.MONGO_URI
+if (config.mongoURI) {
+  mongoURI = config.mongoURI
 } else {
```
Comment (nathan-torquato) [inline backend/db/connectDb.ts]: Unnecessary validation, you already did that in the config file.

File: backend/middleware/authMiddleware.ts [hunk 4]
```diff
@@ -1,11 +1,12 @@
...
+import { UserModel } from '../models/userModel'
+import { config } from '../config'
 
 let JWT_SECRET: string
-if (process.env.JWT_SECRET) {
-  JWT_SECRET = process.env.JWT_SECRET
+if (config.jwtSecret) {
+  JWT_SECRET = config.jwtSecret
 } else {
   throw new Error('JWT_SECRET environment variable is not set')
```
Comment (nathan-torquato) [inline backend/middleware/authMiddleware.ts]: Unnecessary validation, you already did that in the config file.

File: backend/seeder.ts [hunk 5]
```diff
@@ -4,7 +4,7 @@ import products from './data/products'
 import { UserModel } from './models/userModel'
 import Product from './models/productModel'
 import Order from './models/orderModel'
-import connectDB from './db/connectDb'
+import { connectDB } from './db/connectDb'
 
 dotenv.config()
```
Comment (nathan-torquato) [inline backend/seeder.ts]: don't forget to remove `dotenv.config` from every file that's not the `config` file.

File: backend/server.ts [hunk 6]
```diff
@@ -21,14 +19,14 @@ app.use('/api/upload', uploadRoutes)
 
 app.use('/uploads', express.static(path.join(__dirname, '/uploads')))
 
-if (process.env.NODE_ENV === 'production') {
+if (config.shouldServeReactApp) {
```
Comment (nathan-torquato) [inline backend/server.ts:22]: What an elegant solution, kudos! 🤣

File: backend/utils/generateToken.ts [hunk 7]
```diff
@@ -1,16 +1,13 @@
 import jwt from 'jsonwebtoken'
+import { config } from '../config'
 
-// TODO: import from config
 let JWT_SECRET: string
-if (process.env.JWT_SECRET) {
-  JWT_SECRET = process.env.JWT_SECRET
+if (config.jwtSecret) {
+  JWT_SECRET = config.jwtSecret
 } else {
```
Comment (nathan-torquato) [inline backend/utils/generateToken.ts]: Unnecessary validation, you already did that in the config file.

File: instructions.md [hunk 8]
```diff
@@ -0,0 +1,58 @@
+# #TODO 1
```
Comment (nathan-torquato) [inline instructions.md]: great idea!

you can name this file `lessons-learned`.

File: backend/middleware/errorMiddleware.ts [hunk 9]
```diff
@@ -21,7 +22,7 @@ const errorHandler = (
   res.status(statusCode)
   res.json({
     message: err.message,
-    stack: process.env.NODE_ENV === 'production' ? null : err.stack
+    stack: config.productionApplication ? null : err.stack
```
Comment (nathan-torquato) [inline backend/middleware/errorMiddleware.ts]: Remember: do not use the value coming from `NODE_ENV` to implement logic. Do the same we did with the `SERVE_REACT_APP` var for this too.

**Review by nathan-torquato — CHANGES_REQUESTED** [review]
See comments.

File: instructions.md (hunk 8 above)
Comment (Mauricio-Natacci) [reply instructions.md]: accepted

File: backend/utils/generateToken.ts (hunk 7 above)
Comment (Mauricio-Natacci) [reply backend/utils/generateToken.ts]: you're right, removed this part!

File: backend/server.ts (hunk 6 above)
Comment (Mauricio-Natacci) [reply backend/server.ts:22]: hahahahaha I learned this elegant way, did you like it?

File: backend/seeder.ts (hunk 5 above)
Comment (Mauricio-Natacci) [reply backend/seeder.ts]: good eyes hahahah removed!

File: backend/middleware/errorMiddleware.ts (hunk 9 above)
Comment (Mauricio-Natacci) [reply backend/middleware/errorMiddleware.ts]: changed to "shouldServeReactApp" as well!

File: backend/middleware/authMiddleware.ts (hunk 4 above)
Comment (Mauricio-Natacci) [reply backend/middleware/authMiddleware.ts]: you're right, removed this part!

File: backend/db/connectDb.ts (hunk 3 above)
Comment (Mauricio-Natacci) [reply backend/db/connectDb.ts]: you're right, removed this part!

File: .eslintrc (hunk 1 above)
Comment (Mauricio-Natacci) [reply .eslintrc]: I wanted use || instead of ?? and this rule complained hahahaha that's why I turned it off, but I removed this rule right now and it's ok, IDK what happened HAHAHA

**Review by nathan-torquato — APPROVED** [review]
